### ISSUE TITLE
Can't use pouchdb with rollup in the browser

### ISSUE THREAD

**Issue by chmelevskij** [pouchdb/pouchdb#8133]
### Issue
When using pouch db with rollup there are few issues which come up to get it working. It seems to be working fine with webpack because of some magic it does.

Issues are the following:
 1. If using `pouchdb` with simply `@rollup/plugin-node-resolve` and `@rollup/plugin-commonjs` `events` package is not being included if bundling for browser.
 2. If I manually install events and set `@rollup/plugin-node-resolve` to `preferBuiltins: false` it builds the package, but it seems that `node_modules/pouchdb/lib/index-browser.es.js` has `global` keyword, which is odd, because when I've tried to build this repository it was using `self`. Linking it with build from repository worked fine though.
3. Another option to make it work is to add `rollup-plugin-node-globals` and `rollup-plugin-node-builtins` which makes it work as expected, even without the need to install `events` separately. For the minified builds this increases the bundle by about 5kb which is not massive.  

This was also pointed out in [svelte issues](https://github.com/sveltejs/svelte/issues/3267)

### Info
- Environment: browser
- Platform: Chrome
- Adapter: IndexedDB
- Server: NA

### Reproduce
I've created repository to reproduce these issues [here](https://github.com/chmelevskij/pouch-build-reproduce)

**Comment by subz390** [comment]
Hasn't this always been the case.  I don't think it's broken, just a matter of configuring Rollup to handle the package.
I'm using `rollup-plugin-node-polyfills` in my `rollup.config.js` I guess it's the same as `rollup-plugin-node-builtins`

```js
// bundle ES modules from node_modules
// https://github.com/rollup/plugins/tree/master/packages/node-resolve
import resolve from '@rollup/plugin-node-resolve'

// Convert CommonJS modules to ES6, so they can be included in a Rollup bundle
// https://github.com/rollup/plugins/tree/master/packages/commonjs
import commonjs from '@rollup/plugin-commonjs'

// Allows the node builtins to be required/imported.
// https://github.com/ionic-team/rollup-plugin-node-polyfills
import nodePolyfills from 'rollup-plugin-node-polyfills'

export default {
  ...
  plugins: [
    ...
    // the order here is important
    resolve({ browser: true}),
    commonjs(),
    nodePolyfills(),
  ],
}
```

**Comment by axel-recolet** [comment]
I have created a package to use PouchDB with Svelte.

1.   npm i pouchdb-sapper

(when I did this module I thought the problem was sapper but it's with svelte, I think).

2.   You also need to install rollup-plugin-node-polyfills and put it in rollup.config.js
      export default { client : { plugins: [ nodePolyfills(), ... ] } }


3.   put this code in scr\routes\index.svelte (for exemple, feel free to choose an auther)
```javascript
     import PouchDB from 'pouchdb-sapper';

	const init = async () => {
		if(typeof process !== 'undefined') return; // sapper stops there 
		const test = new PouchDB('test');
	}
	init();
```

Look in your browser/indexeddb, a db is creating.

This module is an adaptation of pouchdb-browser@7.2.1  currently 7.2.2
